nl_server/registry.py

```diff
@@ -82,16 +93,7 @@ def _load_models(self, models: dict[str, ModelConfig]):
 
       # try creating a model object from the model info
       try:
-        if model_config.type == ModelType.VERTEXAI:
-          if model_config.usage == ModelUsage.EMBEDDINGS:
-            model = VertexAIEmbeddingsModel(model_config)
-            self.name_to_emb_model[model_name] = model
-          elif model_config.usage == ModelUsage.RERANKING:
-            model = VertexAIRerankingModel(model_config)
-            self.name_to_rank_model[model_name] = model
-        elif model_config.type == ModelType.LOCAL:
-          model = LocalSentenceTransformerModel(model_config)
-          self.name_to_emb_model[model_name] = model
+        self.name_to_emb_model[model_name] = create_model(model_config)
```

Comment:
if `model_config.usage == ModelUsage.RERANKING`, model needs to be added to `self.name_to_rank_model` instead

Reply:
nice catch. updated to keep only one map now.